tensorboard/webapp/widgets/data_table/filter_dialogue.ts

```diff
@@ -0,0 +1,56 @@
+/* Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {
+  DomainType,
+  DiscreteFilter,
+  IntervalFilter,
+  DiscreteFilterValue,
+} from './types';
+import {RangeValues} from '../range_input/types';
+
+@Component({
+  selector: 'tb-data-table-filter',
+  templateUrl: 'filter_dialogue.ng.html',
+  styleUrls: ['filter_dialogue.css'],
+})
+export class FilterDialogue {
+  DomainType = DomainType;
+
+  discreteValueFilter: string = '';
+
+  @Input() filter!: DiscreteFilter | IntervalFilter;
+
+  @Output() discreteFilterChanged = new EventEmitter<DiscreteFilterValue>();
+
+  @Output() intervalFilterChanged = new EventEmitter<RangeValues>();
+
+  @Output() includeUndefinedToggled = new EventEmitter<void>();
+
+  getPossibleValues() {
+    const values: DiscreteFilterValue[] =
+      (this.filter as DiscreteFilter).possibleValues ?? [];
+    if (!this.discreteValueFilter) {
+      return values;
+    }
+    return values.filter((value) =>
+      value.toString().match(this.discreteValueFilter)
+    );
+  }
+
+  discreteValueKeyUp(event: KeyboardEvent) {
+    this.discreteValueFilter = (event.target! as HTMLInputElement).value;
+  }
```

Comment:
This is how I have seen the filter input used before but this is kinda janky. I'm considering
 * Doing this in the template to avoid a somewhat pointless function
 * Modifying the filter input component to either emit the input fields value or support the dreaded two way binding